src/client/pages/Embark/index.tsx

```diff
@@ -215,6 +216,39 @@ export const EmbarkRoot: React.FunctionComponent<EmbarkRootProps> = (props) => {
 
   const textKeys = useTextKeys()
 
+  const trackPassageData = (
+    eventName: string,
+    payload: {
+      [key: string]: any
+    },
+  ) => {
+    const filteredArray = Object.entries(payload).filter(
+      (property) => property[1] !== undefined,
+    )
+
+    const payloadObject = filteredArray.reduce((prev, item) => {
+      const [key, value] = item
+      return { ...prev, [key]: value }
+    }, {})
```

Comment:
If you wanna get fancy:

```suggestion
		const payloadObject = { ...payload }
    Object.keys(payloadObject).forEach(
    	(key) => payloadObject[key] === undefined && delete payloadObject[key],
  	)
```